### PR TITLE
de-interfac-ify Subscriber

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java [hunk 1]
```diff
@@ -388,7 +388,7 @@ public void testFailedChannel_recoverableError_channelReopened() throws Exceptio
     }
 
     final int expectedChannelCount =
-        Runtime.getRuntime().availableProcessors() * SubscriberImpl.CHANNELS_PER_CORE;
+        Runtime.getRuntime().availableProcessors() * Subscriber.CHANNELS_PER_CORE;
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java]: Shouldn't the test `SubscriberImplTest` be renamed to `SubscriberTest`?

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java [hunk 2]
```diff
@@ -85,33 +103,278 @@
...
+    }
+    try {
+      connectionsStopping.await();
+    } catch (InterruptedException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  public SubscriberStats getStats() {
+    // TODO: Implement me
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java]: Do we have this tracked in an issue yet?

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java (hunk 1 above)
Comment (pongad) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java]: The name conflicts with the generated unit test for SubscriberClient.

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java (hunk 2 above)
Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java]: Good idea: https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1542

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java]: Add a to do to fix this.

**Comment by garrettjonesgoogle** [comment]
LGTM

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java (hunk 1 above)
Comment (pongad) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java]: Done: https://github.com/googleapis/toolkit/issues/938

**Pull request merged**